src/java/main/org/apache/zookeeper/server/persistence/FileTxnLog.java

```diff
@@ -311,6 +325,9 @@ public synchronized void commit() throws IOException {
 
                 syncElapsedMS = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startSyncNS);
                 if (syncElapsedMS > fsyncWarningThresholdMS) {
+                    if(serverStats != null) {
+                        serverStats.incrementFsyncThresholdExceedCount();
+                    }
```

Comment:
@nkalmar I wonder if we could add an 'else' branch here with a `LOG.warn()` message indicating that something had gone wrong and Zk was unable to track fsync warnings. What do you think?

Reply:
We could, sure. This nullcheck is just an extra fail safe measurement. It should be never null actually, so if it is, that's definitely some error. (If it would be possible, serverStats would come from constructor actually).